prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -44,21 +60,32 @@ def _get_sensor_data(self, host: str, username: str, password: str) -> Optional[
         redfish_obj: Optional[HttpClient] = None
         try:
             redfish_obj = redfish.redfish_client(
-                base_url=host, username=username, password=password
+                base_url=host,
+                username=username,
+                password=password,
+                timeout=REDFISH_CLIENT_TIMEOUT,
+                max_retry=REDFISH_CLIENT_MAX_RETRY,
             )
             redfish_obj.login(auth="session")
             sensors = redfish_utilities.get_sensors(redfish_obj)
             return sensors
-        except (InvalidCredentialsError, SessionCreationError, ConnectionError) as err:
+        except (
+            InvalidCredentialsError,
+            SessionCreationError,
+            ConnectionError,
+            RetriesExhaustedError,
+        ) as err:
             logger.exception(err)
         finally:
             # Log out
             if redfish_obj:
                 redfish_obj.logout()
         return sensors
 
+    @cached(cache=TTLCache(maxsize=CACHE_MAXSIZE, ttl=CACHE_TTL))
     def discover(self) -> bool:
         """Return true if redfish is been discovered."""
+        logger.info("Discovering redfish services")
```

Comment:
**nitpick (non-blocking):** 

Can we also put log to show the discover result?

Reply:
Good idea. I'll also add another info log in case no redfish services have been discovered.